modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -332,6 +353,31 @@ TensorMechanicsAction::actOutputGeneration()
   }
 }
 
+void
+TensorMechanicsAction::actGatherActionParameters()
+{
+  //
+  // Gather info about all other master actions when we add variables
+  //
+  if (_current_task == "validate_coordinate_systems" && getParam<bool>("add_variables"))
+  {
+    auto actions = _awh.getActions<TensorMechanicsAction>();
+    for (const auto & action : actions)
+    {
+      const auto size_before = _subdomain_id_union.size();
+      const auto added_size = action->_subdomain_ids.size();
+      _subdomain_id_union.insert(action->_subdomain_ids.begin(), action->_subdomain_ids.end());
```

Comment:
I'm doing three things here:

1. creating a set union
2. making sure the original sets are not overlapping
3. making sure none of the sets in the union is empty

I'm open to suggestions on how to make this more compact with STL.